bcgov/HOUS-permit-portal
# Feature/bphh 1510/jurisdiction field map api

## Description
- Add back-end model and data migration to add `JurisdictionIntegrationRequirementsMapping`
- Adds backend call back logic to:
    - Initializes mapping and copies over existing mappings from other templates when a new `JurisdictionIntegrationRequirementsMapping` instance is created
    - Creates mappings for any template version which already does not have a mapping, when api is enabled in jurisdiction
    - Creates missing mapping for any Jurisdiction with api enabled for the template version when the template version in published
    - Syncs common mappings between templates of the same jurisdiction
 - Adds `GET {base_rote}/template_versions/:id/jurisdictions/:jurisdiction_id/jurisdiction_integration_requirements_mapping` and `PATCH {base_rote}/jurisdiction_integration_requirements_mapping`


## What type of PR is this? (check all applicable)

- [x ] 🍕 Feature

## Related Tickets & Documents
https://hous-bssb.atlassian.net/browse/BPHH-1510
https://hous-bssb.atlassian.net/browse/BPHH-1511


## Steps to QA
- Run `rails db:migrate:with_data`
- Ensure there is a published template version
- Update any Jurisdiction to enable the api. `@jurisdiction.update(external_api_enabled: true`
- Run `bundle exec rspec` to verify call back specs are passing
- You should be able to make the following requests via post man now:
<img width="1732" alt="Screenshot 2024-06-06 at 3 06 01 PM" src="https://github.com/bcgov/HOUS-permit-portal/assets/32022335/b937e9f5-7b3b-4a7d-b71e-5759a56f9983">

<img width="1731" alt="Screenshot 2024-06-06 at 4 55 26 PM" src="https://github.com/bcgov/HOUS-permit-portal/assets/32022335/3cfc496e-0bcf-4c14-83f3-58e6ecdda3af">
